### PR TITLE
Fix warning messages

### PR DESCRIPTION
**Pull request by christopheranderson**
Fixes https://github.com/Azure/azure-webjobs-sdk-script/issues/2232

### CODE REVIEW

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 1]
```diff
@@ -419,7 +419,7 @@ public void Initialize()
                     }));
 
                 _eventSubscriptions.Add(EventManager.OfType<HostRestartEvent>()
-                    .Subscribe((msg) => ScheduleRestartAsync(false)));
+                    .Subscribe((msg) => ScheduleRestartAsync(false).ContinueWith(t => _startupLogger.LogError(t.Exception.Message), TaskContinuationOptions.OnlyOnFaulted)));
```
Comment (christopheranderson) [inline src/WebJobs.Script/Host/ScriptHost.cs]: @fabiocav this is the only non-styling impacting changed, besides the gRPC build change

File: src/WebJobs.Script.Grpc/WebJobs.Script.Grpc.csproj [hunk 2]
```diff
@@ -22,10 +22,24 @@
...
+    <Folder Include="Messages\DotNet\" />
+  </ItemGroup>
+
   <Target Name="GenerateProtoFiles" BeforeTargets="BeforeBuild" Inputs="Proto/FunctionRpc.proto" Outputs="Messages/DotNet/FunctionRpc.cs;Messages/DotNet/FunctionRpcGrpc.cs">
     <Exec Command="./generate_protos.$(Ext)" />
     <ItemGroup>
       <Compile Include="**/*$(DefaultLanguageSourceExtension)" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);$(BaseIntermediateOutputPath)**;$(BaseOutputPath)**;@(Compile)" />
     </ItemGroup>
   </Target>
+
```
Comment (christopheranderson) [inline src/WebJobs.Script.Grpc/WebJobs.Script.Grpc.csproj]: This is adding the warning suppressing message to the top of all the gRPC generated C# files.

File: src/WebJobs.Script/Host/ScriptHost.cs (hunk 1 above)
Comment (fabiocav) [reply src/WebJobs.Script/Host/ScriptHost.cs]: 1. Look for an overload (or extension) that logs the exception... we don't want to lose the stack.
2. Continue this synchronously.

Comment (fabiocav) [reply src/WebJobs.Script/Host/ScriptHost.cs]: e.g. https://github.com/fabiocav/azure-webjobs-sdk-script/blob/116b3e7703e0a806c498b9948e511b4b922b5d61/src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs#L218

File: src/WebJobs.Script/Description/DotNet/DotNetMuxer.cs [hunk 3]
```diff
@@ -63,9 +63,9 @@ private static string TryFindMuxerPath()
             }
 
             var muxerDir = new FileInfo(fxDepsFile) // Microsoft.NETCore.App.deps.json
-                .Directory? // (version)
-                .Parent? // Microsoft.NETCore.App
-                .Parent? // shared
+                .Directory?// (version)
```
Comment (fabiocav) [inline src/WebJobs.Script/Description/DotNet/DotNetMuxer.cs]: nit: Don't like the recommendation in this instance... let's add a global suppression instead

File: test/WebJobs.Script.Tests.Shared/TestLoggerProvider.cs [hunk 4]
```diff
@@ -12,21 +12,22 @@ namespace Microsoft.WebJobs.Script.Tests
...
         private readonly Func<string, LogLevel, bool> _filter;
-        private Dictionary<string, TestLogger> _loggerCache { get; } = new Dictionary<string, TestLogger>();
 
         public TestLoggerProvider(Func<string, LogLevel, bool> filter = null)
         {
             _filter = filter;
         }
 
-        public IEnumerable<TestLogger> CreatedLoggers => _loggerCache.Values;
+        private Dictionary<string, TestLogger> LoggerCache { get; } = new Dictionary<string, TestLogger>();
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Shared/TestLoggerProvider.cs]: Odd that this was a private property to begin with

**Pull request merged**